### ISSUE TITLE
javac returns TypeError: Cannot read property '1' of null. 

### ISSUE THREAD

**Issue by izackhub** [NativeScript/nativescript-cli#3152]
Hi developers.
I am having an issue running the android command.
running `tns run android` outputs:
```
`Error executing command 'javac'. Make sure you have installed The Java Development Kit (JDK) and set JAVA_HOME environment variable.`
```

This is a snippet from my `tns docter --log trace` command:
```
<code>
Exec "/Library/Java/Home/bin/javac" -version 
 stdout: javac 9
 
 stderr: 
Command "/Library/Java/Home/bin/javac --version" failed: TypeError: Cannot read property '1' of null
Error executing command 'javac'. Make sure you have installed The Java Development Kit (JDK) and set JAVA_HOME environment variable.
Error: Error executing command 'javac'. Make sure you have installed The Java Development Kit (JDK) and set JAVA_HOME environment variable.
    at Errors.fail (/usr/local/lib/node_modules/nativescript/lib/common/errors.js:125:28)
    at Errors.failWithoutHelp (/usr/local/lib/node_modules/nativescript/lib/common/errors.js:133:14)
    at AndroidToolsInfo.printMessage (/usr/local/lib/node_modules/nativescript/lib/android-tools-info.js:168:26)
    at AndroidToolsInfo.<anonymous> (/usr/local/lib/node_modules/nativescript/lib/android-tools-info.js:113:22)
    at Generator.next (<anonymous>)
    at /usr/local/lib/node_modules/nativescript/lib/android-tools-info.js:13:71
    at Promise (<anonymous>)
    at __awaiter (/usr/local/lib/node_modules/nativescript/lib/android-tools-info.js:9:12)
    at AndroidToolsInfo.validateJavacVersion (/usr/local/lib/node_modules/nativescript/lib/android-tools-info.js:96:16)
    at AndroidProjectService.<anonymous> (/usr/local/lib/node_modules/nativescript/lib/services/android-project-service.js:86:42)
Initializing analytics statuses.
</code>
```
I am using: 
```
`OSX` 10.12.6. 
`node` v8.6.0 
`npm` 5.4.2. 
`tns` 3.2.1
`javac` 9
```
I get an output when I run `javac` without an absolute path. 

I have also run through with `tns doctor` and installed everything, but no change in error msg.

**Comment by NickIliev** [comment]
Hey @izackhub the reason for your issue is because you have Java 9 installed where you should use Java 8 (requirements [here](https://docs.nativescript.org/start/ns-setup-os-x#system-requirements)).  Try installing and setting Java SE Runtime Environment 8u144 from [here ](http://www.oracle.com/technetwork/java/javase/downloads/jre8-downloads-2133155.html)

The support for Java 9 is under development and is planned for the upcoming releases.

**Comment by izackhub** [comment]
Hi @NickIliev thanks for the quick reply. It did indeed solve the problem.
But i have not been able to actually proceed with development due to problems starting any virtual device, so I'm gonna take some time off until I get the time to figure out what is wrong or hopefully gets fix by some update. 

That's not a `NativeScript` issue, but if you have any input on how to find out why a virtual device won't start, I would greatly appreciate it. 

Thank you in advance and have a great week.

**Comment by NickIliev** [comment]
@izackhub you can give some more input on what is happening when trying to start an AVD. I have created [this article](https://docs.nativescript.org/tooling/android-virtual-devices) which demonstrates how to create and use AVD via Android Studio but do let me know if you have some specific issue in your environment.

Closing this issue as the original problem is solved but I will keep track on it if you post more information about your AVD troubles.

**Comment by izackhub** [comment]
Hi @NickIliev 

One of the problem is I don't get any output or error. Everything Installs fine. I followed that exact article to install the AVDs. I tried installing more stuff that I needed. Removed and reinstalled all the ADK and AVD and any additional stuff, incl. Android studio itself. Nothing seems to work.

When I select an AVD and click play, for a moment the OSX dock seems to be expanding for a new icon to appear, but suddenly contracts back after a split second without any msg.

When I try running an application, Android Studio just states "Waiting for target device to come online".
I also tried running "tns run android". It says no device connected.

I need a way to display any error msg that might not be showing. Any proposals?

**Comment by NickIliev** [comment]
@izackhub it is hard to say what causes your case. However, there are few thing you can check out.
- see [this post](https://stackoverflow.com/questions/34503769/trouble-starting-a-avd-android-virtual-device-in-android-studio-on-mac/34512338) and verify that you do not have conflicting virtual machines (e.g. Docker)
- use HAXM and verify that your Mac has enough free resources to start an AVD (see [this post](https://stackoverflow.com/questions/32044251/unable-to-launch-android-emulator-in-mac))

**Comment by izackhub** [comment]
Hi again @NickIliev I managed to fix the error without a lot of hustle. It turns out my $ANDROID_HOME was different from the SDK path that was set in Android Studio. 

I replaced the SDK path in Android Studio, reinstalled all the SDK and AVD, re-cached and restarted and it worked after that. I'm not sure why I had 2 different paths, but it worked.

PS: Before changing the path and reinstalling everything, I removed all installed devices and SDK since you will be reinstalling everything in the new location. Side not for any one who might face the same issue and finds this thread.

Anyway thanks a lot for your time and help. I can finally proceed 👍
